jolie/src/jolie/runtime/typing/TypePair.java

```diff
@@ -0,0 +1,84 @@
+package jolie.runtime.typing;
+
+import jolie.lang.NativeType;
+import jolie.runtime.Value;
+import jolie.util.Range;
+
+import java.util.Map;
+
+/**
+ * Created by User on 01.08.2015.
+ */
+public class TypePair extends Type {
+    public Type getT1() {
+        return t1;
+    }
+    public Type getT2() {
+        return t2;
+    }
+
+    private final Type t1;
+    private final Type t2;
+
+    public TypePair(Type t1, Type t2) {
+        this.t2 = t2;
+        this.t1 = t1;
+    }
+
+    @Override
+    public void cutChildrenFromValue(Value value) {
+
+    }
+
+    @Override
+    public NativeType nativeType() {
+        return null;
+    }
+
+    @Override
+    public Range cardinality() {
+        return null;
+    }
+
+    @Override
+    public Map<String, Type> subTypes() {
+        return null;
+    }
+
+    @Override
+    protected void check(Value value, StringBuilder pathBuilder) throws TypeCheckingException {
+        Boolean isValid;
+        isValid = isCheck(this.getT1(), value, pathBuilder) || isCheck(this.getT2(), value, pathBuilder);
+        if (!isValid){
+            throw new TypeCheckingException( "Invalid types for node " + pathBuilder.toString() + ": expected " + this.getT1().nativeType() + " or " +  this.getT2().nativeType() + ", found " + (( value.valueObject() == null ) ? "void" : value.valueObject().getClass().getName()) );
```

Comment:
How can you be sure that the error came from a native type? Couldn't it be a nested sub node?